test/gtest/ucm/malloc_hook.cc

```diff
@@ -30,49 +30,45 @@ extern "C" {
 
 class malloc_hook : public ucs::test {
 protected:
+    static void mem_event_callback(ucm_event_type_t event_type, ucm_event_t *event,
+            void *arg) {
+        malloc_hook *self = reinterpret_cast<malloc_hook*>(arg);
+        self->m_got_event = 1;
+    }
+
     virtual void init() {
-        size_t free_space, min_free_space, prev_free_space, alloc_size;
-        struct mallinfo mi;
+        ucs_status_t status;
 
+        m_got_event = 0;
         ucm_malloc_state_reset(128 * 1024, 128 * 1024);
         malloc_trim(0);
-
-        /* Take up free space so we would definitely get mmap events */
-        min_free_space  = SIZE_MAX;
-        alloc_size      = 0;
-        mi = mallinfo();
-        prev_free_space = mi.fsmblks + mi.fordblks;
-
-        while (alloc_size < (size_t)(prev_free_space * 0.90)) {
-            m_pts.push_back(malloc(small_alloc_size));
-            alloc_size += small_alloc_size;
-        }
+        status = ucm_set_event_handler(UCM_EVENT_VM_MAPPED,
+                0, mem_event_callback,
+                reinterpret_cast<void*>(this));
+        ASSERT_UCS_OK(status);
 
         for (;;) {
             void *ptr = malloc(small_alloc_size);
-            mi = mallinfo();
```

Comment:
which mallinfo is invoked with upstream? should be the one from ucm/ptmalloc286 if malloc hooks are installed